hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/DynamoDBMetadataStore.java

```diff
@@ -647,9 +650,19 @@ public DirListingMetadata listChildren(final Path path) throws IOException {
           LOG.trace("Listing table {} in region {} for {} returning {}",
               tableName, region, path, metas);
 
-          return (metas.isEmpty() && dirPathMeta == null)
-              ? null
-              : new DirListingMetadata(path, metas, isAuthoritative,
+          if (!metas.isEmpty() && dirPathMeta == null) {
+            // We handle this case as the directory is deleted.
+            LOG.warn("Directory metadata is null, but the list of the "
```

Comment:
we need to review a message which makes sense for someone who doesn't know the db structures. And match it with a troubleshooting doc stack trace.

Reply:
Something like:
> Directory marker is deleted, but the list of the directory elements is not empty: {}. This case is handled as the directory is deleted.
It would be better?